compiler/rustc_span/src/lib.rs

```diff
@@ -1194,6 +1194,31 @@ impl SourceFileHash {
     }
 }
 
+#[derive(Copy, PartialEq, PartialOrd, Clone, Ord, Eq, Hash, Debug, Encodable, Decodable)]
+pub enum DebuggerVisualizerType {
+    Natvis,
+}
+
+rustc_data_structures::impl_stable_hash_via_hash!(DebuggerVisualizerType);
+
+/// A single debugger visualizer file.
+#[derive(Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Encodable, Decodable)]
+pub struct DebuggerVisualizerFile {
+    /// The complete debugger visualizer source.
+    pub src: Lrc<String>,
```

Comment:
Is there are reason for parsing the visualizers into a UTF-8 string? I think it might be more robust to just load them as opaque bytes (e.g. `Vec<u8>` in this case).

Reply:
The main reason is to allow for calling `normalize_src`. This removes the new lines as well as the bom which allows for better deduplicating visualizer files with the same contents, lessening the amount of visualizer files that need to be embedded in the generated PDB/ELF.

Reply:
Is that a problem that would actually occur in practice? It would only make a difference if two included files solely differ in the kind of newlines they use or if one has a BOM and the other doesn't. I think it is unlikely that that happens. 

However, it could be surprising to someone that the compiler would modify the contents of their NatVis files.

Reply:
I understand. I'll make the change to load visualizer files as opaque bytes and not try to normalize the source string since it is rather unlikely to happen.